backend/src/storage.ts

```diff
@@ -62,6 +64,99 @@ export const readLines = async (
 	}
 };
 
+export const createTableDB = async (table: string, attributes: string): Promise<void> => {
+	return new Promise((resolve, reject) => {
+		const db = new Database(DATABASE_PATH);
+		db.serialize(() => {
+			db.run(`CREATE TABLE IF NOT EXISTS ${table} ${attributes}`, (err) => {
+				if (err) {
+					console.error('Error creating the table:', err.message);
+					reject(err);
+					return;
+				}
+				resolve();
+			});
+		});
+		db.close((err) => {
+			if (err) {
+				console.error('Error closing the database:', err.message);
+				reject(err);
+			}
+		});
+	});
+};
+
+export const flushDB = async (table: string): Promise<void> => {
+	return new Promise((resolve, reject) => {
+		const db = new Database(DATABASE_PATH);
+		db.serialize(() => {
+			db.run(`DELETE FROM ${table}`, (err) => {
+				if (err) {
+					console.error('Error creating the table:', err.message);
+					reject(err);
+					return;
+				}
+				resolve();
+			});
+		});
+		db.close((err) => {
+			if (err) {
+				console.error('Error closing the database:', err.message);
+				reject(err);
+			}
+		});
+	});
+};
+
+export const insertDB = async (table: string, values: Map<string, any>): Promise<void> => {
+	return new Promise((resolve, reject) => {
+		const db = new Database(DATABASE_PATH);
+		db.serialize(() => {
+			const columns = Array.from(values.keys()).join(', ');
+			const placeholders = Array.from(values.keys())
+				.map(() => '?')
+				.join(', ');
+			const stmt = db.prepare(`INSERT INTO ${table} (${columns}) VALUES (${placeholders})`);
+			stmt.run(Array.from(values.values()), (err) => {
+				if (err) {
+					console.error('Error inserting into the database:', err.message);
+					reject(err);
+					return;
+				}
+			});
+			stmt.finalize();
+			resolve();
+		});
+		db.close((err) => {
+			if (err) {
+				console.error('Error closing the database:', err.message);
+				reject(err);
+			}
+		});
+	});
+};
+
+export const readDB = async (query: string) => {
```

Comment:
Okay it makes sense that this wouldn't be factorized as you need more flexibility for reading, but in that case the function should be named queryDB (oui je fais chier sur des détails y'a quoi)

Reply:
Détail à la con but ok, ig

Reply:
Après en l'appelant queryDB tu peux t'en servir pour plein d'autres trucs genre supprimer des valeurs etc. sans que ça soit bizarre 🤷‍♂️